scripts/start_ray.py

```diff
@@ -41,16 +42,25 @@ def check_no_existing_redis_clients(node_ip_address, redis_address):
     # Start Ray on the head node.
     if args.redis_address is not None:
       raise Exception("If --head is passed in, a Redis server will be started, so a Redis address should not be provided.")
+
     # Get the node IP address if one is not provided.
     if args.node_ip_address is None:
       node_ip_address = services.get_node_ip_address()
     else:
       node_ip_address = args.node_ip_address
     print("Using IP address {} for this node.".format(node_ip_address))
-    address_info = services.start_ray_local(node_ip_address=node_ip_address,
-                                            num_workers=args.num_workers,
-                                            cleanup=False,
-                                            redirect_output=True)
+
+    if args.redis_port is not None:
+      address_info_in = {"redis_address": "{}:{}".format(node_ip_address,
```

Comment:
I'd prefer to call it address_info instead of address_info_in. I'm aware the name is also used below but it is not a problem, as it is explicitly updated like so:
address_info = function(address_info)